syncto/tests/test_functional.py

```diff
@@ -59,6 +59,44 @@ def test_404_endpoint_returns_cors_headers(self):
         self.assertEqual(response.headers['Access-Control-Allow-Origin'],
                          'notmyidea.org')
 
+    def test_client_state_header_with_non_sense_raise_401(self):
+        headers = self.headers.copy()
+        headers['Authorization'] = "BrowserID abcd"
+        headers['X-Client-State'] = "NonSense"
+        patch = mock.patch("syncto.authentication.SyncClient")
+        with patch as sync_client:
+            error = HTTPError()
+            error.response = mock.MagicMock()
+            error.response.status_code = 401
+            error.response.reason = "Unauthorized"
+            error.response.text = ('{"status": "invalid-credentials", '
+                                   '"errors": [{"location": "body", '
+                                   '"name": "", '
+                                   '"description": "Unauthorized"}]}')
+            sync_client.side_effect = error
+            resp = self.app.get(COLLECTION_URL, headers=headers, status=401)
+
+        self.assertFormattedError(
+            resp, 401, ERRORS.INVALID_AUTH_TOKEN, "Unauthorized",
+            '401 Unauthorized: {"status": "invalid-credentials"')
+
+    def test_error_with_syncclient_server_raise_a_503(self):
+        headers = self.headers.copy()
+        headers['Authorization'] = "BrowserID abcd"
+        headers['X-Client-State'] = "NonSense"
+        patch = mock.patch("syncto.authentication.SyncClient")
+        with patch as sync_client:
+            error = HTTPError()
+            error.response = mock.MagicMock()
+            error.response.status_code = 503
+            error.response.reason = "Service Unavailable"
+            sync_client.side_effect = error
+            resp = self.app.get(COLLECTION_URL, headers=headers, status=503)
+
+        self.assertFormattedError(
+            resp, 503, ERRORS.BACKEND, "Service Unavailable",
+            "Service unavailable due to high load, please retry later.")
```

Comment:
We should read this directly from Cliquet rather than testing it here. In case it evolves, the tests will be broken otherwise.